spread/client.go

```diff
@@ -281,26 +281,32 @@ func (c *Client) run(script string, dir string, env *Environment, mode outputMod
 		rebootKey = rerr.Key
 		output = append(output, '\n')
 
-		timedout := time.After(c.killTimeout)
-		err := c.Run(fmt.Sprintf("reboot &\nsleep %.0f", c.killTimeout.Seconds()), "", nil)
+		uptime, err := c.getUptime()
 		if err != nil {
-			err = c.Run("echo should-have-disconnected", "", nil)
+			return nil, fmt.Errorf("failed to get uptime before reboot, error: %v", err)
```

Comment:
This error message should be inside getUptime (with some tuning, will comment there).